### DISCUSSION TITLE
Dashboard file-server from sd-card (esp32-s3)

### DISCUSSION THREAD

**Discussion by pirzimm** [cesanta/mongoose#2470] [Ask questions, get answers] [answered]
Hello everyone,

I would like to build a file server with mongoose so that it is possible to download log files from the external sd card of the esp32-s3 via the dashboard.
I am currently running a slightly modified version of the dashboard example on mg version 7.12.
Is there an example that does something similar? 
I tried to use the mg_http_serve_dir function, but I think this is not compatible with the external SD card.
(I'm not that experienced with frontend implementation and http communication protocols, which is why I'm stuck)

Thanks for any help

**Comment by scaprile** [comment, marked as answer]
Please follow the guidelines in the [documentation](https://mongoose.ws/documentation),  [examples](https://github.com/cesanta/mongoose/tree/master/examples) and [tutorials](https://mongoose.ws/tutorials/) available.

Mongoose is not compatible nor not-compatible with your external SD or whatever, it reads a file system. The very first lines in `app_main()` in the example mount a filesystem. Make sure the one you want is mounted.
https://github.com/cesanta/mongoose/tree/master/examples/esp32/device-dashboard/main/main.c#L10-L18
Then, to serve files from that filesystem, read the docs and add the proper filesystem option to `mg_http_serve_dir()`:
https://mongoose.ws/documentation/#filesystem
